File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>old;
        old=nums;
        sort(nums.begin(),nums.end());
        int start=0;
        int end=nums.size()-1;
        int ans1=0;
        int ans2=0;
        while(start<end){
            if((nums[start]+nums[end])==target){
               ans1=nums[start];
               ans2=nums[end];
               break;
            }
            else if(nums[start]+nums[end]<target){
                start++;
            }
            else{
                end--;
            }
        }
        vector<int>finalans;
        for(int i=0;i<old.size();i++){
            if(ans1==old[i]){
              finalans.push_back(i);
            }
            else if(ans2==old[i]){
                finalans.push_back(i);
            }
        }
        return finalans;
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] old = Arrays.copyOf(nums, nums.length);
        Arrays.sort(nums);
        int start = 0;
        int end = nums.length - 1;
        int ans1 = 0;
        int ans2 = 0;
        while (start < end) {
            if ((nums[start] + nums[end]) == target) {
                ans1 = nums[start];
                ans2 = nums[end];
                break;
            } else if (nums[start] + nums[end] < target) {
                start++;
            } else {
                end--;
            }
        }
        List<Integer> finalAnsList = new ArrayList<>();
        for (int i = 0; i < old.length; i++) {
            if (ans1 == old[i]) {
                finalAnsList.add(i);
            } else if (ans2 == old[i]) {
                finalAnsList.add(i);
            }
        }
        // Convert List<Integer> to int[]
        int[] finalAns = new int[finalAnsList.size()];
        for (int i = 0; i < finalAnsList.size(); i++) {
            finalAns[i] = finalAnsList.get(i);
        }
        return finalAns;
    }
}